config

```diff
@@ -1,937 +1,854 @@
+    } elsif ( $GUESSOS =~ '.*-aix' ) {
+        $KERNEL_BITS //= `(getconf KERNEL_BITMODE) 2>/dev/null`;
+        $KERNEL_BITS //= '32';
+        my $OBJECT_MODE //= 32;
+        if ( $CC eq "gcc" ) {
+            $OUT = "aix-gcc";
+            if ( $OBJECT_MODE == 64 ) {
+                print 'Your $OBJECT_MODE was found to be set to 64';
+                $OUT = "aix64-gcc"
+            }
+            elsif ( $OBJECT_MODE == 64 ) {
+                print 'Your $OBJECT_MODE was found to be set to 64';
+                $OUT = "aix64-cc";
+            } else {
+                $OUT = "aix-cc";
+            }
+            if ( $KERNEL_BITS eq '64' ) {
+                print <<EOF;
+WARNING! To build 64-bit kit, inovke
+         $THERE/Configure aix64-cc
+EOF
+                maybe_abort();
+            }
+        }
+        if ( okrun(
+                "lsattr -E -O -l `lsdev -c processor|awk '{print \$1;exit}'`"
+                . "| grep -i powerpc) >/dev/null 2>&1") ) {
+            # this applies even to Power3 and later, as they return
+            # PowerPC_POWER[345]
+        } else {
+            $options .= " no-asm";
+        }
+    } elsif ( $GUESSOS =~ 'i[3456]86-.*-cygwin' ) {
+        return "Cygwin-x86";
+    } elsif ( $GUESSOS =~ '.*-.*-cygwin' ) {
+        return "Cygwin-${MACHINE}";
+    } elsif ( $GUESSOS =~ 'x86-.*-android|i.86-.*-android' ) {
+        return "android-x86";
+    } elsif ( $GUESSOS =~ 'armv[7-9].*-.*-android' ) {
+        $__CNF_CFLAGS .= " -march=armv7-a";
+        $__CNF_CXXFLAGS .= " -march=armv7-a";
+        return "android-armeabi";
+    } elsif ( $GUESSOS =~ 'arm.*-.*-android' ) {
+        return "android-armeabi";
+    } else {
+        # Last cause, return "z" from x-y-z
+        my @fields = split(/-/, $GUESSOS);
+        return $fields[2];
+    }
+    # In case we didn't return above.
+    return $OUT;
+}
+
+$OUT = map_guess();
+
+$options .= " $CONFIG_OPTIONS" if $CONFIG_OPTIONS ne '';
+
+## gcc < 2.8 does not support -march=ultrasparc
+if ( $OUT eq 'solaris-sparcv9-gcc' && $GCCVER < 28 ) {
+    print <<EOF;
+WARNING! Downgrading to solaris-sparcv8-gcc
+         Upgrade to gcc-2.8 or later.
+EOF
+  maybe_abort();
+  $OUT = 'solaris-sparcv8-gcc';
+}
+if ( $OUT eq "linux-sparcv9" &&a $GCCVER < 28 ) {
+  print <<EOF;
+WARNING! Downgrading to linux-sparcv8
+         Upgrade to gcc-2.8 or later.
+EOF
+  maybe_abort();
+  $OUT = 'linux-sparcv8';
+}
+
+$OUT //= $CC;
+
+##
+## FIND PERL VERSION 5
+sub find_perl_5 {
+    return $PERL if $PERL ne '';
+    foreach my $dir ( split(/:/, $ENV{PATH}) ) {
+        my $d = $dir eq '' ? '.' : $dir;
+        return "$d/perl5${EXE}" if -x "$d/perl5${EXE}";
+    }
+    foreach my $dir ( split(/:/, $ENV{PATH}) ) {
+        my $d = $dir eq '' ? '.' : $dir;
+        return "$d/perl${EXE}"
+            if -x "$d/perl${EXE}" && okrun("$d/perl${EXE} -e 'exit($]<5.0);");
+    }
+    die "You need Perl 5." if $PERL eq '';
+}
+
+$PERL = &find_perl_5();
```

Comment:
This is quite unnecessary, since we have the variable `$^X`. See https://perldoc.perl.org/perlvar.html

Reply:
are you saying remove those lines and do 
```
$PERL = $^X;
```
?

Reply:
Remove them, their only purpose is to find a PERL to run Configure in, but since you're going to convert this to a module that Configure can `use`, those lines are pointless.

Either way, it's really silly to go search for perl when you're running a perl script, since the information is already there...

Reply:
done.